### PR TITLE
[MPS] Fix MPS sorting issue with strided view tensors

### LINKED ISSUE #101878: torch.quantile on MPS doesn't sort values when dim is not None
### 🐛 Describe the bug

MPS implementation of `torch.quantile` doesn't sort the input when there's the `dim` argument is not `None`, giving nonsense results.

```
import torch
device = torch.device("mps")
A = torch.randn(1000, 10)

#need to use "nearest", "lower", or "higher" interpolation methods
#"linear" and "midpoint" fail due to operator `aten::lerp.Tensor_out` not being implemented for MPS
#when dim=None, CPU and MPS give same result
torch.quantile(A, .5, interpolation = "nearest")
torch.quantile(A.to(device), .5, interpolation = "nearest")

#when dim is not None, MPS gives nonsense result
torch.quantile(A, .5, 0, interpolation = "nearest")
torch.quantile(A.to(device), .5, 0, interpolation = "nearest")

#MPS output is same as middle row of tensor
#so torch.quantile must just be pulling 500th index without sorting first
A[500]

#inelegant workaround is to sort the tensor before calling quantile
torch.quantile(torch.sort(A.to(device), dim = 0).values, .5, 0, interpolation = "nearest")
```

### Versions

PyTorch version: 2.0.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.3.1 (x86_64)
GCC version: Could not collect
Clang version: 14.0.6
CMake version: version 3.22.1
Libc version: N/A

Python version: 3.11.3 (main, Apr 19 2023, 18:51:09) [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-10.16-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M2 Pro

Versions of relevant libraries:
[pip3] numpy==1.24.3
[pip3] torch==2.0.1
[pip3] torchmetrics==0.11.4
[conda] nomkl                     3.0                           0  
[conda] numpy                     1.24.3          py311hd0e4f0d_0  
[conda] numpy-base                1.24.3          py311h947b413_0  
[conda] torch                     2.0.1                    pypi_0    pypi
[conda] torchmetrics              0.11.4                   pypi_0    pypi

### PR DESCRIPTION
**Pull request by Az0s**
Fixes #101878

The issue with the `sort_stable_out_mps` function is that it does not handle strided views of the input tensor correctly. Specifically, if the input tensor is a strided view, the function will produce incorrect results.

This fix ensures that the input tensor is contiguous before passing it to the mps calculation. test case also added. let me know if this is a proper solution

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/Sort.mm [hunk 1]
```diff
@@ -29,10 +29,15 @@
...
   MPS_CHECK_INT64_OP_SUPPORTED(self, macOS13_3_plus, "sort_stable_out");
 
-  values.copy_(self);
-  // check if self is scalar
-  dim = maybe_wrap_dim(dim, self.dim(), true);
-  if (self.dim() == 0 && self.numel() == 1) {
+
+  // issue #101878: MPS might give a wrong sorted tensor when dealing with a strided view tensor.
+  // create a proxy tensor to temporarily address the issue
+  Tensor input_contiguous = self.contiguous();
```
Comment (kulinseth) [inline aten/src/ATen/native/mps/operations/Sort.mm:35]: Making it contiguous will have perf implications. can you try similar to :
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/mps/operations/BinaryOps.mm#L84

see if this still works

Comment (Az0s) [reply aten/src/ATen/native/mps/operations/Sort.mm:35]: appreciate it! looking into it for a more thorough solution

**Review by kulinseth — APPROVED** [review]

File: aten/src/ATen/native/mps/operations/Sort.mm (hunk 1 above)
Comment (kulinseth) [reply aten/src/ATen/native/mps/operations/Sort.mm]: Thanks

**Comment by Az0s** [comment]
Hello @kulinseth ,

I've been closely following the progress of this PR and noticed that it has received reviews but has not yet been merged. Could this be due to the CI checks not passing?

Upon reviewing the CI logs, I found that the `test_output_match_nanquantile_cpu_float32` test, which was initially expected to fail, has `unexpectedly succeeded`. If this CI issue needs to be addressed within this PR, could you please advise on the best course of action? Should the test now be marked as expected to pass?

thanks for the time and consideration
